internal/adapters/rest/controller/order_controller.go

```diff
@@ -48,62 +83,61 @@ func (ctrl OrderController) GetAll(c *gin.Context) {
 // @Accept json
 // @Produce json
 // @Success 200 {object} order.Order{}
-// @Failure 400 {object} map[string]any "{\"error\": \Invalid identifier informed\"}"
-// @Router /order/{id} [get]
-func (ctrl OrderController) Get(c *gin.Context) {
+// @Failure 400 {object} HTTPError
+// @Failure 422 {object} HTTPError
+// @Failure 500 {object} HTTPError
+// @Router /orders/{id} [get]
+func (ctrl *OrderController) Get(c *gin.Context) {
 	id, err := uuid.Parse(c.Param("id"))
 
 	if err != nil {
-		c.JSON(http.StatusBadRequest, gin.H{
-			"error": "Invalid identifier informed.",
-		})
-
+		ctrl.Error(c, errutil.NewInputError(err))
 		return
 	}
 
 	o, err := ctrl.service.Get(id)
-
 	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{
-			"error": err.Error(),
-		})
+		ctrl.Error(c, err)
 
 		return
 	}
 
 	c.JSON(http.StatusOK, o)
 }
 
-type CreateOrderPayload struct {
-	ClientID uuid.UUID `json:"client_id" binding:"required" format:"uuid"`
+type UpdateOrderPayload struct {
+	Status string `json:"status" binding:"required" example:"paid"`
 }
 
-// Create Order godoc
-// @Summary Create order from Cart
-// @Description Create order from Cart
-// @Param payload body controller.CreateOrderPayload true "CreateOrderPayload"
+// Update Order godoc
+// @Summary Update an Order
+// @Description Update by json an Order
+// @Param id path string true "ID of the Order"
+// @Param payload body controller.UpdateOrderPayload true "UpdateOrderPayload"
 // @Tags Order
 // @Accept  json
 // @Produce  json
 // @Success 200 {object} order.Order{}
-// @Failure 500 {object} map[string]any "{\"error\": \Internal Server Error\"}"
-// @Router /order [post]
-func (ctrl OrderController) Create(c *gin.Context) {
-	payload := &CreateOrderPayload{}
+// @Failure 400 {object} HTTPError
+// @Failure 422 {object} HTTPError
+// @Failure 500 {object} HTTPError
+// @Router /orders/:id [patch]
```

Comment:
Só um ajuste pra agrupar com a outra rota de mesmo endpoint na documentação. Também notei que esse recurso em específico retorna erro de CORS ao testar pela documentação do swagger (mesmo com middleware de CORS aplicado), os demais recursos funcionam corretamente.

```suggestion
// @Router /orders/{id} [patch]
```

Reply:
Eita, tô acostumado com a semântica do Fiber 😅